### ISSUE TITLE
audio_send_stream wrong

### ISSUE THREAD

**Issue by zengchuan** [devopvoid/webrtc-java#85]
In dev: win10 with soundcard, all is right. (pc1)browser(addtrack)->server(webrtc-java ontrack->addtrack)->browser(ontrack pc2).
In production: ubuntu18.0.4 
```
apt-get install pulseaudio
adduser root pulse-access
rm -rf /var/run/pulse /var/lib/pulse /root/.config/pulse
pulseaudio -D --verbose --exit-idle-time=-1 --system --disallow-exit
pactl list
Sink #0
        State: SUSPENDED
        Name: alsa_output.pci-0000_00_05.0.analog-stereo
        Description: Built-in Audio Analog Stereo
        Driver: module-alsa-card.c
        Sample Specification: s16le 2ch 44100Hz
        Channel Map: front-left,front-right
        Owner Module: 1
        Mute: yes
        Volume: front-left: 293 /   0% / -140.98 dB,   front-right: 293 /   0% / -140.98 dB
                balance 0.00
        Base Volume: 6554 /  10% / -60.00 dB
        Monitor Source: alsa_output.pci-0000_00_05.0.analog-stereo.monitor
        Latency: 0 usec, configured 0 usec
        Flags: HARDWARE HW_MUTE_CTRL HW_VOLUME_CTRL DECIBEL_VOLUME LATENCY
        Properties:
                alsa.resolution_bits = "16"
                device.api = "alsa"
                device.class = "sound"
                alsa.class = "generic"
                alsa.subclass = "generic-mix"
                alsa.name = "Intel 82801AA-ICH"
                alsa.id = "Intel ICH"
                alsa.subdevice = "0"
                alsa.subdevice_name = "subdevice #0"
                alsa.device = "0"
                alsa.card = "0"
                alsa.card_name = "Intel 82801AA-ICH"
                alsa.long_card_name = "Intel 82801AA-ICH with STAC9700,83,84 at irq 21"
                alsa.driver_name = "snd_intel8x0"
                device.bus_path = "pci-0000:00:05.0"
                sysfs.path = "/devices/pci0000:00/0000:00:05.0/sound/card0"
                device.bus = "pci"
                device.vendor.id = "8086"
                device.vendor.name = "Intel Corporation"
                device.product.id = "2415"
                device.product.name = "82801AA AC'97 Audio Controller"
                device.form_factor = "internal"
                device.string = "front:0"
                device.buffering.buffer_size = "17632"
                device.buffering.fragment_size = "4408"
                device.access_mode = "mmap"
                device.profile.name = "analog-stereo"
                device.profile.description = "Analog Stereo"
                device.description = "Built-in Audio Analog Stereo"
                alsa.mixer_name = "SigmaTel STAC9700,83,84"
                alsa.components = "AC97a:83847600"
                module-udev-detect.discovered = "1"
                device.icon_name = "audio-card-pci"
        Ports:
                analog-output;output-amplifier-on: Analog Output / Amplifier (priority: 9910)
                analog-output;output-amplifier-off: Analog Output / No Amplifier (priority: 9900)
        Active Port: analog-output;output-amplifier-on
        Formats:
                pcm

Source #0
        State: SUSPENDED
        Name: alsa_output.pci-0000_00_05.0.analog-stereo.monitor
        Description: Monitor of Built-in Audio Analog Stereo
        Driver: module-alsa-card.c
        Sample Specification: s16le 2ch 44100Hz
        Channel Map: front-left,front-right
        Owner Module: 1
        Mute: no
        Volume: front-left: 65536 / 100% / 0.00 dB,   front-right: 65536 / 100% / 0.00 dB
                balance 0.00
        Base Volume: 65536 / 100% / 0.00 dB
        Monitor of Sink: alsa_output.pci-0000_00_05.0.analog-stereo
        Latency: 0 usec, configured 0 usec
        Flags: DECIBEL_VOLUME LATENCY
        Properties:
                device.description = "Monitor of Built-in Audio Analog Stereo"
                device.class = "monitor"
                alsa.card = "0"
                alsa.card_name = "Intel 82801AA-ICH"
                alsa.long_card_name = "Intel 82801AA-ICH with STAC9700,83,84 at irq 21"
                alsa.driver_name = "snd_intel8x0"
                device.bus_path = "pci-0000:00:05.0"
                sysfs.path = "/devices/pci0000:00/0000:00:05.0/sound/card0"
                device.bus = "pci"
                device.vendor.id = "8086"
                device.vendor.name = "Intel Corporation"
                device.product.id = "2415"
                device.product.name = "82801AA AC'97 Audio Controller"
                device.form_factor = "internal"
                device.string = "0"
                module-udev-detect.discovered = "1"
                device.icon_name = "audio-card-pci"
        Formats:
                pcm

Source #1
        State: SUSPENDED
        Name: alsa_input.pci-0000_00_05.0.analog-stereo
        Description: Built-in Audio Analog Stereo
        Driver: module-alsa-card.c
        Sample Specification: s16le 2ch 44100Hz
        Channel Map: front-left,front-right
        Owner Module: 1
        Mute: no
        Volume: front-left: 27636 /  42% / -22.50 dB,   front-right: 27636 /  42% / -22.50 dB
                balance 0.00
        Base Volume: 27636 /  42% / -22.50 dB
        Monitor of Sink: n/a
        Latency: 0 usec, configured 0 usec
        Flags: HARDWARE HW_MUTE_CTRL HW_VOLUME_CTRL DECIBEL_VOLUME LATENCY
        Properties:
                alsa.resolution_bits = "16"
                device.api = "alsa"
                device.class = "sound"
                alsa.class = "generic"
                alsa.subclass = "generic-mix"
                alsa.name = "Intel 82801AA-ICH"
                alsa.id = "Intel ICH"
                alsa.subdevice = "0"
                alsa.subdevice_name = "subdevice #0"
                alsa.device = "0"
                alsa.card = "0"
                alsa.card_name = "Intel 82801AA-ICH"
                alsa.long_card_name = "Intel 82801AA-ICH with STAC9700,83,84 at irq 21"
                alsa.driver_name = "snd_intel8x0"
                device.bus_path = "pci-0000:00:05.0"
                sysfs.path = "/devices/pci0000:00/0000:00:05.0/sound/card0"
                device.bus = "pci"
                device.vendor.id = "8086"
                device.vendor.name = "Intel Corporation"
                device.product.id = "2415"
                device.product.name = "82801AA AC'97 Audio Controller"
                device.form_factor = "internal"
                device.string = "front:0"
                device.buffering.buffer_size = "17632"
                device.buffering.fragment_size = "4408"
                device.access_mode = "mmap"
                device.profile.name = "analog-stereo"
                device.profile.description = "Analog Stereo"
                device.description = "Built-in Audio Analog Stereo"
                alsa.mixer_name = "SigmaTel STAC9700,83,84"
                alsa.components = "AC97a:83847600"
                module-udev-detect.discovered = "1"
                device.icon_name = "audio-card-pci"
        Ports:
                analog-input-mic;input-microphone-1: Microphone / Microphone 1 (priority: 8720)
                analog-input-mic;input-microphone-2: Microphone / Microphone 2 (priority: 8719)
                analog-input-linein: Line In (priority: 8100)
                analog-input-aux: Analog Input (priority: 8000)
                analog-input-video: Video (priority: 7000)
        Active Port: analog-input-mic;input-microphone-1
        Formats:
                pcm

```
video is right but audio is wrong. errors:
```
# Fatal error in: ../../audio/audio_send_stream.cc, line 387
# last system error: 1
# Check failed: !race_checker.RaceDetected()
```

**Comment by devopvoid** [comment]
Hi, I'm not sure how I can help you here. I suppose this has nothing to do with the Java JNI implementation.

Please consult the WebRTC Group: https://groups.google.com/g/discuss-webrtc

**Comment by devopvoid** [comment]
Fix with #118
